cmake/compiler.cmake

```diff
@@ -276,19 +271,6 @@ macro(enable_tnt_compile_flags)
         endif()
     endif()
 
-    if (CMAKE_COMPILER_IS_CLANG AND CC_HAS_WNO_UNUSED_VALUE)
-        # False-positive warnings for ({ xx = ...; x; }) macroses
-        add_compile_flags("C;CXX" "-Wno-unused-value")
-    endif()
-
-    if (CC_HAS_WNO_CHAR_SUBSCRIPTS)
-        add_compile_flags("C;CXX" "-Wno-char-subscripts")
-    endif()
```

Comment:
It seems `CC_HAS_WNO_CHAR_SUBSCRIPTS` (defined on line 95) itself becomes unneeded. The same likely relevant for other such variables.

Reply:
Thanks for the suggestion, removed all such variables. 

We also have some unused compiler flag checks, but I will leave them for now.